Updating Read me (#16)
Azure-Samples/ms-identity-ciam-native-auth-ios-sample: README.md

# Sign in users and call a protected web API in iOS (Swift) mobile app by using native authentication

* [Overview](#overview)
* [Contents](#contents)
* [Prerequisites](#prerequisites)
* [Project setup](#project-setup)
* [Key concepts](#key-concepts)
* [Reporting problems](#reporting-problems)
* [Contributing](#contributing)

## Overview

This sample iOS sample applications demonstrates how to handle sign-up, sign-in, sign out, and reset password scenarios using Microsoft Entra External ID for customers. You can configure the sample to call a protected web API.

## Contents

> | File/folder | Description |

Reviewer: the "[NativeAuthSampleApp.xcodeproj](https://github.com/Azure-Samples/ms-identity-ciam-native-auth-ios-sample/tree/main/NativeAuthSampleApp.xcodeproj)" is still missing from this list

    static let tenantSubdomain = "Enter_the_Tenant_Subdomain_Here"
}
```

The code creates two constant properties:

* _clientId_ - the value _Enter_the_Application_Id_Here_ is replaced with **Application (client) ID** of the app you register during the project setup. The **Application (client) ID** is unique identifier of your registered application.
* _tenantSubdomain_ - the value _Enter_the_Tenant_Subdomain_Here_ is replaced with the Directory (tenant) subdomain. The tenant subdomain URL is used to construct the authentication endpoint for your app.

You use `NativeAuthSampleApp/Configuration.swift` file to set configuration options when you initialize the client app in the Microsoft Authentication Library (MSAL).

To create SDK instance, use the following code:

```swift
import MSAL

var nativeAuth: MSALNativeAuthPublicClientApplication!

do {
    nativeAuth = try MSALNativeAuthPublicClientApplication(
        clientId: Configuration.clientId,
        tenantSubdomain: Configuration.tenantSubdomain,
        challengeTypes: [.OOB, .password]
    )
} catch {
    print("Unable to initialize MSAL \(error)")
    showResultText("Unable to initialize MSAL")
}
```

> You create MSAL instance so that you can perform authentication logic and interact with your tenant through native authentication APIs. The `MSALNativeAuthPublicClientApplication` create an instance called `nativeAuth`. The configuration file `NativeAuthSampleApp/Configuration.swift` file is passed as parameter.For more information about SDK instance, see [Tutorial: Prepare your iOS app for native authentication](https://learn.microsoft.com/en-gb/entra/external-id/customers/tutorial-native-authentication-prepare-ios-app#create-sdk-instance)

Reviewer: ```suggestion
You create MSAL instance so that you can perform authentication logic and interact with your tenant through native authentication APIs. The `MSALNativeAuthPublicClientApplication` create an instance called `nativeAuth`. The `clientId` and `tenantSubdomain`, defined in theconfiguration file `NativeAuthSampleApp/Configuration.swift` file, are passed as parameter. For more information about SDK instance, see [Tutorial: Prepare your iOS app for native authentication](https://learn.microsoft.com/en-gb/entra/external-id/customers/tutorial-native-authentication-prepare-ios-app#create-sdk-instance)
```
